### DISCUSSION TITLE
Plan to replace ElasticSearch with BanyanDB in quick start and showcase/demo on 2024 Q1/Q2

### DISCUSSION THREAD

**Discussion by wu-sheng** [apache/skywalking#11886] [Ideas]
With BanyanDB 0.6 coming in next month, SkyWalking's BanyanDB team will announce the first official performance test results for the medium deployment scale of using agent and Istio ALS. This deployment scale should be satisfied for most of our users.

To help people get familiar with this new database, we are going to provide a new quick-start shell(https://github.com/apache/skywalking/issues/11619) to install OAP + BanyanDB easily on local env, most likely using local docker-compose. Meanwhile, SkyWalking showcase and its default deployment(demo env, a.k.a. demo.skywalking.a.o) are planning to switch to running with BanyanDB, rather than ElasticSearch.

The final performance report would show the benefits of lower resource costs. The DAY2(Operation and maintenance capability) will be noted after 0.6 is released, and is the first class consideration when moving toward 1.0.

I am posting this discussion to provide a heads-up for the whole community. BanyanDB storage option would change to **RECOMMENDED TO TRY** mode from **FOR-DEV-ONLY**(alpha) mode.

**Comment by wu-sheng** [comment]
FYI @apache/skywalking-committers

**Comment by lujiajing1126** [comment]
Glad to see a big step forward after 3yrs!
I believe this would help us make the product better with both stability and performance.

**Comment by Superskyyy** [comment]
Love to see it happen!

**Comment by SooryaMuthuraj** [comment]
Waiting for it!

**Comment by innerpeacez** [comment]
Nice job!

**Comment by john1337** [comment]
good news，cause elasticsearch worked not so well after running a long time

**Reply by wu-sheng** [reply]
That is not true. Elasticsearch works well, the reason for the replacement is due to efficiency only.

**Comment by jmjoy** [comment]
I look forward to it.

**Comment by wu-sheng** [comment]
Quick start updated by @kezhenxu94  through https://github.com/apache/skywalking/pull/12014

It depends on BanyanDB 0.6 release and OAP v10. FYI @hanahmily @wankai123 

Demo and showcase will update too.

**Comment by wu-sheng** [comment]
Officially, BanyanDB 0.6 is ready to release. We plan to do that this week. OAP will adopt that version by using the latest released 0.6.0 client.

**Comment by wu-sheng** [comment]
BanyanDB 0.6.1 and OAP 10.0.1 are being deployed into demo as showcase has been updated, https://github.com/apache/skywalking-showcase/pull/171
